### ISSUE TITLE
Align data model for reports with Vega (change from map to list of objects)

### ISSUE THREAD

**Issue by jonrkarr** [reproducible-biomedical-modeling/CRBM-Viz#2299]
The current data model below for the values of data sets of SED reports (e.g., columns of CSV result tables) isn't immediately compatible with Vega.
```typescript
type SimulationRunReportData =
  | { [label: string]: Array<number> }
  | { [label: string]: Array<boolean> };
```

Because it's not compatible with Vega, I needed to transform our data for use with Vega (this is done with TypeScript code in the dispatch app). This means that we can't use URLs to our API to connect Vega with our data. Instead, the values of the data have to be retrieved externally to the Vega viewer and embedded into the description of the visualization. This isn't a huge problem, but it could be cleaner. The ability to use our URLs would make it a little easier to work with the Vega ecosystem, including the Vega editors.

### Alternative data model
Basically, the data model just needs to be changed from a map of labels to values to an array of objects of labels and values:
```typescript
interface SimulationRunReportDatum {
  id: string;  
  label: string;
  values: number[] | boolean[];
}

type SimulationRunReportData = SimulationRunReportDatum[];
```

This has the additional advantage of enabling the data structure to capture the id of each SED-ML data set, as well as its label. I think design is also a bit more conventional for data storage systems.

**Comment by jonrkarr** [comment]
The output data model has been aligned such that API URLs can now be used directly in Vega documents.
